test/sslapitest.c

```diff
@@ -10349,25 +10372,28 @@ static int test_pipelining(int idx)
     size_t written, readbytes, offset, msglen, fragsize = 10, numpipes = 5;
     size_t expectedreads;
     unsigned char *buf = NULL;
-    ENGINE *e;
-
-    if (!TEST_ptr(e = ENGINE_by_id("dasync")))
-        return 0;
+    ENGINE *e = NULL;
 
-    if (!TEST_true(ENGINE_init(e))) {
-        ENGINE_free(e);
-        return 0;
+    if (idx != 6) {
+        e = load_dasync();
+        if (e == NULL)
+            return 0;
     }
 
-    if (!TEST_true(ENGINE_register_ciphers(e)))
-        goto end;
-
     if (!TEST_true(create_ssl_ctx_pair(libctx, TLS_server_method(),
                                        TLS_client_method(), 0,
                                        TLS1_2_VERSION, &sctx, &cctx, cert,
                                        privkey)))
         goto end;
 
+    if (idx == 6) {
+        e = load_dasync();
+        if (e == NULL)
+            goto end;
+        /* Now act like test 0 */
+        idx = 0;
```

Comment:
Nit: not too keen on this, as it might make debugging the test harder; but... since the value is not printed out via debug or anything, it may not matter.

Reply:
Actually, I'm not even sure this is needed, as there are no explicit checks for `idx == 0` or `idx == 6`

Reply:
We do similar things in numerous other tests, so I don't think this is a big problem. The alternative is to add additional complexity to the test so that everywhere we do something for test 0 we do the same thing for test 7. I think that is worse.

Reply:
In this particular case, it doesn't seem to matter (see my comment at the same time as yours).